### ISSUE TITLE
Abnormal pilot-agent CPU usage if DR certs are not found

### ISSUE THREAD

**Issue by LithiumSR** [istio/istio#44986]
### Bug Description

Hello everyone,
I found an issue that affects Istio 1.16+.
From what I found out, since the backoff changes performed in b63f73e an abnormal CPU usage  shows up under certain scenarios.
I can reproduce the issue when I define a DR that uses certs that do not exist inside the sidecar.

I have some simple steps to reproduce the issue:
1. Install Istio 1.17.2 (or 1.16.4 if you prefer) with `istioctl install`
2. Deploy the repro application ([link](https://www.dropbox.com/s/8tigs1s5tqiqx74/repro-chart-0.1.0.tgz?dl=0) to helm chart) using `helm install repro repro-chart-0.1.0.tgz -n istio-system`
The repro application contains the following:
- Sleep client that will be injected with sidecar. Sleep has a sidecar.istio.io/userVolume annotation to mount secrets inside the sidecar.
- A SE + DR that defines the path where to find the certs for TLS. Keep in mind that the path used will not match any file inside the sidecar.
- Secrets definition with empty content

3. Due to the fact that secrets are empty, and the certs cannot be found inside the sidecar,
istio-proxy will rightfully raise some errors/warnings about this

```
2023-05-18T21:45:47.576398Z     info    Envoy proxy is ready
2023-05-18T21:45:51.865837Z     error   cache   resource:file-root:/etc/istio/egress-ca-certs/ca-chain.cert.pem failed to generate secret for proxy from file: open /etc/istio/egress-ca-certs/ca-chain.cert.pem: no such file or directory
2023-05-18T21:45:51.866054Z     info    ads     ADS: "@" sleep-669f4db5dc-k4lth.istio-system-3 terminated
2023-05-18T21:45:51.866330Z     warning envoy config external/envoy/source/common/config/grpc_stream.h:163      StreamSecrets gRPC config stream to sds-grpc closed: 2, failed to generate secret for file-root:/etc/istio/egress-ca-certs/ca-chain.cert.pem: open /etc/istio/egress-ca-certs/ca-chain.cert.pem: no such file or directory  thread=26
2023-05-18T21:45:52.198401Z     info    ads     ADS: new connection for node:sleep-669f4db5dc-k4lth.istio-system-6
```

4. Check that the CPU usage of pilot-agent is normal. I use the "top" command inside the sidecar container to verify this.
```
Tasks:   3 total,   1 running,   2 sleeping,   0 stopped,   0 zombie
%Cpu(s):  3.0 us,  2.3 sy,  0.0 ni, 94.3 id,  0.1 wa,  0.0 hi,  0.3 si,  0.0 st
MiB Mem :  32083.0 total,  14394.2 free,   2966.8 used,  14722.0 buff/cache
MiB Swap:      0.0 total,      0.0 free,      0.0 used.  28652.6 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                      
     24 istio-p+  20   0  181020  55716  38880 S   0.3   0.2   0:01.50 envoy                                                                                                        
      1 istio-p+  20   0  755004  47512  31740 S   0.0   0.1   0:00.80 pilot-agent                                                                                                  
     39 istio-p+  20   0    7292   3476   2936 R   0.0   0.0   0:00.02 top     
```                                                                                                     

5. After exactly 15 minutes from the deployment of sleep the CPU usage will abruptly rise to abnormal levels.

```
top - 22:01:17 up 22 days,  3:58,  0 users,  load average: 0.95, 0.64, 0.54
Tasks:   3 total,   1 running,   2 sleeping,   0 stopped,   0 zombie
%Cpu(s): 10.2 us, 12.7 sy,  0.0 ni, 76.7 id,  0.0 wa,  0.0 hi,  0.4 si,  0.0 st
MiB Mem :  32083.0 total,  13623.4 free,   3119.3 used,  15340.4 buff/cache
MiB Swap:      0.0 total,      0.0 free,      0.0 used.  28500.1 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                      
      1 istio-p+  20   0  755004  55712  33084 S 193.0   0.2   1:29.97 pilot-agent                                                                                                  
     26 istio-p+  20   0  183004  57452  38768 S   0.3   0.2   0:04.10 envoy                                                                                                        
     39 istio-p+  20   0    7292   3492   2952 R   0.0   0.0   0:00.28 top    
```

6. If you were to replace the empty secret with a secret that has data then the CPU usage will go back to normal

### Version

```Text
client version: 1.17.2
control plane version: 1.17.2
data plane version: 1.17.2 (2 proxies)
```

### Additional Information

[bug-report.tar.gz captured during the abnormal cpu usage](https://github.com/istio/istio/files/11511946/bug-report.tar.gz)

### Affected product area

- [ ] Ambient
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [X] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

**Comment by LithiumSR** [comment]
Tagging @hzxuzhonghu since it seems related to b63f73e28706f2cb78b7ed85f3646731a6c11e87

**Comment by hzxuzhonghu** [comment]
I am not sure why backoff should be related

**Comment by hzxuzhonghu** [comment]
I can reproduce, and strace shows pilot-agent is reading cert files

**Comment by hzxuzhonghu** [comment]
https://github.com/istio/istio/pull/44993

**Comment by LithiumSR** [comment]
Thank you a lot @hzxuzhonghu for taking care of it 😊

**Comment by hzxuzhonghu** [comment]
can you test it, i manually tested. Works for me

**Comment by LithiumSR** [comment]
Sure, I will test right away

**Comment by kfaseela** [comment]
thank you so much for the fix @hzxuzhonghu :)

**Comment by LithiumSR** [comment]
Hello @hzxuzhonghu,
Sorry for the delay, I can confirm that the issue is no longer there after applying the fix 👍 
Thank you again

**Comment by LithiumSR** [comment]
I have verified that the commit merged as part of PR #44993 fixed the issue.
I will close the issue.
Thank you again 😃
